File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        a = 0
        b = 0
        count = defaultdict(int)
        maxCount = 0
        longest = 0
        while b < len(s):
            count[s[b]] += 1
            maxCount = max(maxCount, count[s[b]])

            while b - a + 1 - maxCount > k:
                count[s[a]] -= 1
                a += 1

            longest = max(longest, b - a + 1)

            b += 1
        return longest

File: Data Structures & Algorithms/climbing-stairs/submission-4.cpp
class Solution {
public:
    int climbStairs(int n) {
        int a = 1;
        int b = 1;
        for (int i = 0; i < n - 1; i++) {
            int t = b;
            b = a + b;
            a = t;
        }
        return b;
    }
};
